app/models/storage_location.rb

```diff
@@ -136,52 +136,25 @@ def remove_empty_items
   end
 
   # FIXME: After this is stable, revisit how we do logging
-  def increase_inventory(itemizable_array)
-    itemizable_array = itemizable_array.to_a
+  def increase_inventory(item_hash_array)
+    change_log = {}
 
-    # This is, at least for now, how we log changes to the inventory made in this call
-    log = {}
-    # Iterate through each of the line-items in the moving box
-    Item.reactivate(itemizable_array.map { |item_hash| item_hash[:item_id] })
-    itemizable_array.each do |item_hash|
-      # Locate the storage box for the item, or create a new storage box for it
-      inventory_item = inventory_items.find_or_create_by!(item_id: item_hash[:item_id])
-      # Increase the quantity-on-record for that item
-      new_quantity = inventory_item.quantity + item_hash[:quantity].to_i
-      inventory_item.update!(quantity: new_quantity)
-      # Record in the log that this has occurred
-      log[item_hash[:item_id]] = "+#{item_hash[:quantity]}"
-    end
+    Item.reactivate(item_hash_array.map { |item| item[:item_id] })
+
+    item_hash_array.each { |item| update_item_quantity(change_log, item, increase: true) }
     # log could be pulled from dirty AR stuff?
     # Save the final changes -- does this need to occur here?
     save
-    # return log
-    log
+
+    change_log
   end
 
   # TODO: re-evaluate this for optimization
-  def decrease_inventory(itemizable_array)
-    itemizable_array = itemizable_array.to_a
+  def decrease_inventory(item_hash_array)
```

Comment:
Not that I think these refactors are bad per se, but this is already a very meaty change. Can we limit the scope of the PR to only what's needed to implement the changes requested?

In addition, the whole inventory thing is due for an overhaul very soon, so this code might be removed in the near future anyway. :smile:

Reply:
Sounds good!